tests/acceptance/pageObjects/FilesPageElement/appSideBar.js

```diff
@@ -54,6 +56,37 @@ module.exports = {
      */
     isCollaboratorsTabPresentOnCurrentSidebar: function() {
       return this.isTabPresentOnCurrentSidebar(this.elements.sidebarCollaboratorsTab.selector)
+    },
+    markFavoriteSidebar: function() {
+      return this.useXpath()
+        .waitForElementVisible('@sideBar')
+        .waitForElementVisible('@favoriteStarDimm')
+        .click('@sidebarToggleFavoriteButton')
+        .waitForElementVisible('@favoriteStarShining')
+    },
+    unmarkFavoriteSidebar: function() {
+      return this.waitForElementVisible('@sideBar')
+        .waitForElementVisible('@favoriteStarShining')
+        .click('@sidebarToggleFavoriteButton')
+        .waitForElementVisible('@favoriteStarDimm')
+    },
+
+    /**
+     * Asserts whether the resource in the sidebar is marked as favorite
+     * @param {Boolean} shouldBeMarked
+     */
+    isMarkedAsFavorite: function(shouldBeMarked) {
```

Comment:
assertions should be done in the context file, the pageObject should only return the result